docs/runbooks.md

```diff
@@ -0,0 +1,140 @@
+
+How to **investigate**:
+
+- Review the `rollout-operator` logs (or trace) to gain insight into what may be causing a delay or blockage in the `zpdb` eviction controller
+- Use caution with restarting the `rollout-operator` pod. It maintains internal state of recently evicted pods
+- Ensure that the `pod-eviction` and `zpdb-validation` `ValidatingWebhookConfiguration` have a failure policy set to `Fail` before restarting the `rollout-operator`
+
+## Metrics
+
+A prometheus metrics service is available at `/metrics` of the `rollout-operator` deployment.
+
+### kube_validating_webhook_failure_policy
+
+This metric reports on the current configuration of validating and mutating webhook failure policy configurations.
+
+Labels are used to indicate the policy mode (`Fail` or `Ignore`) and the webhook details. A value of `1` indicates that this is the current setting.
+
+Use this metric to monitor that the webhooks have been correctly configured.
+
+### rollout_operator_zpdb_configurations_observed_total
+
+This counter metric reports on the total number of `ValidatingWebhookConfiguration` configurations which have been `updated` (including additions) or `deleted`.
+
+It also tracks the number of `ignored` and `invalid` configuration updates.
+
+A configuration will be `ignored` if it is a stale update or an update in an unexpected format.
+
+A configuration will be `invalid` if it fails a validation process.
+
+Use this metric to monitor for changes to the `zpdb` configurations and to monitor for unexpected `invalid` configurations being observed. This may indicate an error in the generation of these configurations.
+
+### rollout_operator_zpdb_eviction_requests_total
+
+This counter metric reports on the total number of pod eviction requests.
+
+This includes both pod eviction requests which come in via the pod eviction webhook, and pod deletion requests which come from the rollout controller (pod updates).
+
+Note that the number of requests arriving via the pod eviction webhook can be tracked with the `rollout_operator_kubernetes_api_client_request_duration_seconds_count` metric.
+
+Use this metric to monitor for abnormal request volume and/or frequency.
+
+### rollout_operator_zpdb_inflight_eviction_requests
+
+This is a gauge metric which tracks the number of in-flight pod eviction requests.
+
+Like `rollout_operator_zpdb_eviction_requests_total`, this metrics takes both eviction requests which come in via the pod eviction webhook, and pod deletion requests which come from the rollout controller (pod updates).
+
+Note that the number of in-flight requests via the pod eviction webhook can be tracked with the `rollout_operator_inflight_requests` metric.
+
+Use this metric to monitor for abnormal high volumes of in-flight requests. Since these eviction requests should return quickly, even a small number of sustained in-flight requests is likely indicative of an error.
```

Comment:
```suggestion
Use this metric to monitor for abnormal high volumes of in-flight requests. Since these eviction requests should return quickly, even a small number of sustained in-flight requests is likely indicative of an issue.
```